contracts/service-registry/src/state.rs

```diff
@@ -300,6 +377,61 @@ pub fn deregister_chains_support(
     Ok(())
 }
 
+fn update_auth_verifier_count_change_tracker(
+    current: &mut i32,
+    previous_state: &Option<AuthorizationState>,
+    auth_state: &AuthorizationState,
+) -> Result<(), ContractError> {
+    match (previous_state, auth_state) {
+        (Some(NotAuthorized) | Some(Jailed), Authorized) | (None, Authorized) => {
+            *current = current
+                .checked_add(1)
+                .ok_or(ContractError::AuthorizedVerifiersIntegerOverflow)?;
+        }
+        (Some(Authorized), NotAuthorized) | (Some(Authorized), Jailed) => {
+            *current = current
+                .checked_sub(1)
+                .expect("authorized verifiers count should not underflow");
+        }
+        _ => {}
+    }
+    Ok(())
+}
```

Comment:
```suggestion
fn calculate_auth_verifier_count_change(
    previous_state: Option<&AuthorizationState>,
    auth_state: &AuthorizationState,
) -> i16 {
    match (previous_state, auth_state) {
        (Some(Authorized), Authorized) => 0,
        (Some(Authorized), _) => -1,
        (_, Authorized) => 1,
        _ => 0
    }
}
```
Try to avoid passing mutable state around. Also &Option<T> clones the underlying type, whereas Option<&T> just copies the reference